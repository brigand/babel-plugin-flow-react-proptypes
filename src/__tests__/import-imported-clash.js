const babel = require('babel-core');
const content = `
import type {Foo} from './foo';
import type {Foo as Bar} from './bar';

type Props = {
  foo: Foo,
  bar: Bar,
};

class C extends React.Component {
    props: Props
};
`;

it('import-imported-clash', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
