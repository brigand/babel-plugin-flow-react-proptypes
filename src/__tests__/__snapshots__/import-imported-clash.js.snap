// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-imported-clash 1`] = `
"'use strict';

var _foo = require('./foo');

var _bar = require('./bar');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var C = function (_React$Component) {
  _inherits(C, _React$Component);

  function C() {
    _classCallCheck(this, C);

    return _possibleConstructorReturn(this, (C.__proto__ || Object.getPrototypeOf(C)).apply(this, arguments));
  }

  return C;
}(React.Component);

C.propTypes = {
  foo: function foo() {
    return (typeof _foo.bpfrpt_proptype_Foo === 'function' ? _foo.bpfrpt_proptype_Foo.isRequired ? _foo.bpfrpt_proptype_Foo.isRequired : _foo.bpfrpt_proptype_Foo : _propTypes2.default.shape(_foo.bpfrpt_proptype_Foo).isRequired).apply(this, arguments);
  },
  bar: function bar() {
    return (typeof _bar.bpfrpt_proptype_Foo === 'function' ? _bar.bpfrpt_proptype_Foo.isRequired ? _bar.bpfrpt_proptype_Foo.isRequired : _bar.bpfrpt_proptype_Foo : _propTypes2.default.shape(_bar.bpfrpt_proptype_Foo).isRequired).apply(this, arguments);
  }
};
;"
`;
